### PR TITLE
*: Update client-go and verify all read ts

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve?

Issue Number: ref #57786

Problem Summary:

* Update client-go to include https://github.com/tikv/client-go/pull/1513 , which tries to validate all read ts.
* Adapt interface chagnes in client-go
* Remove some redundant checks that can be done in client-go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/set.go [hunk 1]
```diff
@@ -222,7 +222,8 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
 	newSnapshotTS := getSnapshotTSByName()
 	newSnapshotIsSet := newSnapshotTS > 0 && newSnapshotTS != oldSnapshotTS
 	if newSnapshotIsSet {
-		err = sessionctx.ValidateSnapshotReadTS(ctx, e.Ctx().GetStore(), newSnapshotTS)
+		isStaleRead := name == variable.TiDBTxnReadTS
```
Comment (MyonKeminta) [inline pkg/executor/set.go:225]: See: https://github.com/pingcap/tidb/pull/57050/files#diff-7b12a7041569f8636f7e0830ec0cba3df3630271c7963fe7f7d5480101fbe295L225-L228

**Review by zyguan — APPROVED** [review]

**Review by ekexium — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

**Review by winoros — APPROVED** [review]
planner part lgtm

File: pkg/ddl/backfilling_operators.go [hunk 2]
```diff
@@ -549,6 +550,8 @@ func (w *tableScanWorker) HandleTask(task TableScanTask, sender func(IndexRecord
 		}
 		w.se = session.NewSession(sessCtx)
 	}
+	ts, err := w.se.GetStore().GetOracle().GetTimestamp(context.Background(), &oracle.Option{TxnScope: oracle.GlobalTxnScope})
+	logutil.BgLogger().Info("***** check ts on se", zap.Uint64("ts", ts), zap.Error(err))
```
Comment (cfzjywxk) [inline pkg/ddl/backfilling_operators.go:554]: Is this debug log?

**Comment by MyonKeminta** [comment]
retest

**Review by tangenta — APPROVED** [review]

**Pull request merged**